barcodes_generator_abstract/models/barcode_rule.py

```diff
@@ -54,6 +61,33 @@ def onchange_generate_type(self):
             if rule.generate_type == 'no':
                 rule.generate_model = False
 
+    # Constrains Section
+    @api.multi
+    @api.constrains('generate_model', 'generate_automate')
+    def _check_generate_model_automate(self):
+        """ It should not allow two automated barcode generators per model.
+
+        It also clears the cache of automated rules if necessary.
+        """
+        should_clear = False
+        for record in self:
+            if not record.generate_automate:
+                continue
+            # This query is duplicated, but necessary because the other
+            # method is cached & we need a completely current result.
+            domain = [
+                ('generate_model', '=', record.generate_model),
+                ('generate_automate', '=', True),
+            ]
+            if len(self.search(domain)) > 1:
+                raise exceptions.ValidationError(_(
+                    'Only one rule per model can be used for automatic '
+                    'barcode generation.'
+                ))
+            should_clear = True
+        if should_clear:
+            self.clear_caches()
```

Comment:
For keeping the same logic, just create a new method that is called from both write/create methods. I still miss the wanted @onwrite decorator: https://github.com/odoo/odoo/pull/11042